### PR TITLE
Create CompatHelper.yml

### PR DESCRIPTION
**Pull request by mforets**
This adds the CompatHelper bot. However, it rmay remain to setup deploy keys (https://github.com/JuliaRegistries/CompatHelper.jl#1-installation), and I don't have privileges in this repo to help with that matter.

### CODE REVIEW

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  CompatHelper:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Pkg.add("CompatHelper")
+        run: julia -e 'using Pkg; Pkg.add("CompatHelper")'
+      - name: CompatHelper.main()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COMPATHELPER_PRIV: ${{ secrets.COMPATHELPER_PRIV }}  # optional
```
Comment (lbenet) [inline .github/workflows/CompatHelper.yml:15]: Do we have to define `COMPATHELPER_PRIV` token or not? If so, could we use `DOCUMENTER_KEY` as in [here](https://github.com/PerezHz/TaylorIntegration.jl/blob/76cd51a807f27bd7a9a51fc96e4a088884f83248/.github/workflows/docs.yml#L27)?

**Review by lbenet — COMMENTED** [review]
Thanks a lot! I left one comment due to (my) ignorance. Aside from that, LGTM!

File: .github/workflows/CompatHelper.yml (hunk 1 above)
Comment (mforets) [reply .github/workflows/CompatHelper.yml:15]: indeed, im no expert but i think you can also remove it 
 see other packages eg .https://github.com/SciML/DifferentialEquations.jl/blob/master/.github/workflows/CompatHelper.yml

Comment (lbenet) [reply .github/workflows/CompatHelper.yml:15]: Can you comment it, so if we require it it is a simple change? Then I merge this!

Comment (mforets) [inline .github/workflows/CompatHelper.yml:15]:
```suggestion
```

**Review by lbenet — APPROVED** [review]
Thanks a lot! I merge this once the tests are finished!

**Comment by lbenet** [comment]
Thanks again @mforets. Merging!

**Pull request merged**